ui/src/lib/gql/job/useSubscriptions.ts

```diff
@@ -0,0 +1,125 @@
+  const wsClient = useWsClient();
+  const queryClient = useQueryClient();
+  const isSubscribedRef = useRef(false);
+  const unSubscribedRef = useRef<(() => void) | undefined>(undefined);
+  const lastStatusRef = useRef<any>(null);
+
+  const query = useQuery({
+    queryKey: [JobQueryKeys.GetJobStatus, jobId],
+    queryFn: async () => {
+      const cachedData = queryClient.getQueryData([
+        JobQueryKeys.GetJobStatus,
+        jobId,
+      ]);
+      if (cachedData) {
+        lastStatusRef.current = cachedData;
+        return cachedData;
+      }
+      return null;
+    },
+    // Important: initial query should run only once
+    staleTime: Infinity,
+    gcTime: Infinity,
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+  });
+
+  useEffect(() => {
+    if (!jobId || isSubscribedRef.current) return;
+
+    console.log(`Setting up job status subscription for job: ${jobId}`);
+    isSubscribedRef.current = true;
+
+    // Initialize with any cached data
+    const cachedData = queryClient.getQueryData([
+      JobQueryKeys.GetJobStatus,
+      jobId,
+    ]);
+    if (cachedData) {
+      lastStatusRef.current = cachedData;
+    }
+
+    // Subscribe to job status
+    const unsubscribe = wsClient.subscribe<OnJobStatusChangeSubscription>(
+      {
+        query: JOB_STATUS_SUBSCRIPTION,
+        variables: { jobId },
+      },
+      {
+        next: (data) => {
+          if (data.data?.jobStatus) {
+            const newStatus = data.data.jobStatus;
+
+            const currentStatus = JSON.stringify(lastStatusRef.current);
+            const incomingStatus = JSON.stringify(newStatus);
+
+            if (currentStatus !== incomingStatus) {
+              console.log(`Job status changed for ${jobId}:`, {
+                previous: lastStatusRef.current,
+                new: newStatus,
+              });
+
+              lastStatusRef.current = newStatus;
+
+              // Update React Query cache
+              queryClient.setQueryData(
+                [JobQueryKeys.GetJobStatus, jobId],
+                toJobStatus(newStatus),
+              );
+            } else {
+              console.log(
+                `Received same status ${newStatus}, no update needed`,
+              );
+            }
+          }
+        },
+        error: (err) => {
+          console.error(`Status subscription error ${jobId}:`, err);
+          isSubscribedRef.current = false;
+        },
+        complete: () => {
+          console.info("Status Subscription complete");
+          isSubscribedRef.current = false;
+        },
+      },
+    );
+
+    unSubscribedRef.current = unsubscribe;
+
+    // Cleanup
+    return () => {
+      unsubscribe();
+      isSubscribedRef.current = false;
+    };
+  }, [jobId, wsClient, queryClient]);
+
+  const stopSubscription = useCallback(async () => {
+    if (unSubscribedRef.current) {
+      unSubscribedRef.current();
```

Comment:
`unSubscribedRef?.current();` is enough. Don't need the if statement (once the isSubscribedRef is removed